bentoml/yatai/yatai_service_impl.py

```diff
@@ -578,6 +584,79 @@ def ContainerizeBento(self, request, context=None):
                     logger.error(f"RPC ERROR ContainerizeBento: {e}")
                     return ContainerizeBentoResponse(status=Status.INTERNAL(e))
 
-            # pylint: enable=unused-argument
+        def UploadBento(self, request_iterator, context=None):
+            try:
+                with TempDirectory() as temp_dir:
+                    temp_tar_path = os.path.join(
+                        temp_dir, f'{uuid.uuid4().hex[:12]}.tar'
+                    )
+                    with open(temp_tar_path, 'wb+') as file:
+                        for request in request_iterator:
+                            if not request.bento_bundle:
+                                bento_name = request.bento_name
```

Comment:
Is the bento_name, bento_version request always the first item in the iterator?

I think a few things we can do after it received the bento name and version:

* check if the bento with name, version is already in the DB, return error if not (you have it down in this function, let's move it up)
* check the upload status(this is missing in the PR right now), if it's already in progress or done, return an error
* acquire the write lock, make sure it's not already being uploaded in another client
* extract the bento files to the right URL
* Set the status to "uploading" with the metadata store API
* Return the response